cmd/auth-rpc-client.go

```diff
@@ -117,15 +119,20 @@ func newAuthClient(cfg *authConfig) *AuthRPCClient {
 
 // Close - closes underlying rpc connection.
 func (authClient *AuthRPCClient) Close() error {
+	authClient.mu.Lock()
 	// reset token on closing a connection
 	authClient.isLoggedIn = false
+	authClient.mu.Unlock()
 	return authClient.rpc.Close()
 }
 
 // Login - a jwt based authentication is performed with rpc server.
 func (authClient *AuthRPCClient) Login() error {
+	authClient.mu.Lock()
```

Comment:
I think better to serialize calls to `Login()` be just doing:

```go
func (authClient *AuthRPCClient) Login() error {
        authClient.mu.Lock()
        defer authClient.mu.Unlock()
        // login code follows.
}
```

No point racing to call the loginMethod from multiple threads, right?